pkg/api/v1/types.go

```diff
@@ -1413,6 +1419,14 @@ type Container struct {
 	// Cannot be updated.
 	// +optional
 	Env []EnvVar `json:"env,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,7,rep,name=env"`
+	// List of sources to populate environment variables in the container.
+	// The keys defined within a source must be a C_IDENTIFIER.
+	// When a key exists in multiple sources, the value associated with the last
+	// source will take precedence.
+	// All env values will take precedence over any listed source.
+	// Cannot be updated.
+	// +optional
+	EnvFrom []EnvFromSource `json:"envFrom,omitempty" protobuf:"bytes,19,rep,name=envFrom"`
```

Comment:
Hmm. We'd need to extend strategic merge patch in order to support this properly. We definitely want automated processes to be able to add additional configmaps (#17097).

One alternative would be to not use the OneOf pattern in EnvFromSource and instead include an object reference that would specify the resource type (kind and apiVersion). Another alternative would be to use separate arrays for configmaps, secrets, services, and any types we add in the future that would define multiple environment variables. The former sounds less objectionable.

Any other ideas @pwittrock?